components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/test/java/org/wso2/carbon/policy/mgt/core/PolicyManagerServiceImplTest.java

```diff
@@ -326,9 +338,38 @@ public void inactivatePolicy() throws PolicyManagementException {
     }
 
     @Test(dependsOnMethods = "updatePolicy")
-    public void deletePolicy() throws PolicyManagementException {
+    public void deletePolicyById() throws PolicyManagementException {
         policyManagerService.deletePolicy(policy1.getId());
         Policy tempPolicy = policyManagerService.getPAP().getPolicy(policy1.getId());
         Assert.assertNull(tempPolicy, "Policy was not deleted successfully");
     }
-}
+
+    @Test(dependsOnMethods = "updatePolicy")
+    public void deletePolicyByPolicy() throws PolicyManagementException {
+        policyManagerService.deletePolicy(policy1);
+        Policy tempPolicy = policyManagerService.getPAP().getPolicy(policy1.getId());
+        Assert.assertNull(tempPolicy, "Policy was not deleted successfully");
+    }
+
+    @Test(dependsOnMethods = "applyPolicy")
+    public void getEffectiveFeatures( ) throws Exception {
+        List<ProfileFeature> effectiveFeatures = policyManagerService.
+                getEffectiveFeatures(new DeviceIdentifier(DEVICE1, DEVICE_TYPE_A));
+        Assert.assertNotNull(effectiveFeatures);
+        Assert.assertEquals(POLICY1_FEATURE1_CODE,effectiveFeatures.get(0).getFeatureCode());
+        try{
+              policyManagerService.getEffectiveFeatures(new DeviceIdentifier(DEVICE_2, DEVICE_TYPE_B));
+        }catch(FeatureManagementException e){
+           Assert.assertTrue(e.getCause()instanceof PolicyEvaluationException);
```

Comment:
Please throw back the exception if the exception cause is not an instance of the PolicyEvaluationException. We need to throw back all exceptions, rather trying to assert the exceptions there, else during the test failures due to unexpected exceptions, we'll not get the full trace.